theano/tensor/nlinalg.py

```diff
@@ -625,9 +625,12 @@ def perform(self, node, inputs, outputs):
         (x,) = inputs
         (w, u, v) = outputs
         assert x.ndim == 2, "The input of svd function should be a matrix."
-        w[0], u[0], v[0] = self._numop(x,
-                                       self.full_matrices,
-                                       self.compute_uv)
+        if self.compute_uv:
+            w[0], u[0], v[0] = self._numop(x,
+                                           self.full_matrices,
+                                           self.compute_uv)
+        else:
+            u[0] = self._numop(x, self.full_matrices, self.compute_uv)
```

Comment:
For the CPU op, as there is no C code, this isn't a problem.

For the GPU op, as it is a props, you can do it. I think this is an old questions that you found the answer.

Can you add a CPU test for that and fix the CPU make_node?